_query-dsl/rewrite-parameter.md

```diff
@@ -0,0 +1,174 @@
+---
+layout: default
+title: Rewrite
+nav_order: 80
+---
+
+# Rewrite
+
+Multi-term queries like `wildcard`, `prefix`, `regexp`, `fuzzy`, and `range` expand internally into sets of terms. The `rewrite` parameter allows you to control how these term expansions are executed and scored.
+
+When a multi-term query expands into many terms (for example `prefix: "error*"` matching hundreds of terms), they are converted into actual term queries internally. This process can:
+
+* Exceed the `indices.query.bool.max_clause_count` limit (default `1024`)
+* Affect how scores are calculated for matching documents
+* Impact memory and latency depending on the rewrite method used
+
+## Available rewrite methods
+
+| Rewrite method | Description |
+| [`constant_score`](#constant_score-default) | (Default) All expanded terms are evaluated together as a single unit, assigning the same score to every match. Efficient for filtering use cases. |
```

Comment:
How does it help with filtering use case, scoring happens on filtered documents right ?

Reply:
If we meant that it will save the scoring cost by passing the scoring phase for all docs then it makes sense

Reply:
@kaushalmahi12 Thank you for the review, could you please check the updated changes?